scripts/strict-null-checks-ci.js

```diff
@@ -0,0 +1,170 @@
+/**
```

Comment:
what is the best way to deal with a script like this? I wrote it in JS to avoid have to deal with some sort of build step before being able to just run it in the action, but we could write it in typescript instead and either execute it with `ts-node` or do a quick `tsc` on it.

My hope is that this is something that only sticks around for a few months before we're able to delete it, so I'm inclined to keep things lightweight and uncomplicated

Reply:
The `scripts` directory normally gets transpiled as a part of the `npm run build` command (we do this on every build to ensure the tooling in `scripts` is always up to date).  If we made this a `.ts` file, it'd get compiled out to `scripts/build/`. That does present a bit of a chicken/egg problem, since we're doing this step outside of the `build` step in CI to get that parallelism.

I think there are 4 options here:

## 1) Make this file a `.ts` file, compile the `scripts` directory as a part of this workflow

Compiling the `scripts` directory is a relatively quick process (compared to type checking the compiler & running rollup on everything).  We could run tsc across the directory (`node scripts --prepare`), then point the job to run `scripts/build/strict-null-checks-ci.js`

This would add a little length of time to the job - the output on my machine for `time node scripts --prepare) is ~5 seconds, and would probably longer in GH's hosted runners (especially the Windows ones 🙃). But, it'd give us type safety "for free".

## 2) Use ts-node

I don't see why this _couldn't_ work. We'd probably need to `npm install` it as a part of the job and make sure its executable - off the top of my head that's the most work we'd have to do.  I'd prefer option 1 over this to keep things 'simple' (i.e. not spin up another way we transpile/run things).

## 3) Use `// @ts-check` directive

We could add the `// @ts-check` directive to the top of the file, and annotate it with types.  It's "nice" in that we don't have to set up any compilation steps, but it's a bit tedious to do and the type checking isn't nearly as strong as using tsc (and editor dependent insofar as how 'loud' you make TS yell at you).  I added an example patch of what this might look like (I had to make it a .txt file, because you can't upload `.patch` files to GH in comments 🤣) - [example-patch.txt](https://github.com/ionic-team/stencil/files/8679049/example-patch.txt)

## 4) Do Nothing/Leave it as is

Probably my least favorite option. My personal preference would be to get type safety from the get-go.